pymatgen/analysis/piezo_sensitivity.py

```diff
@@ -0,0 +1,602 @@
+        if not (obj - np.transpose(obj, (0, 1, 3, 2)) < tol).all():
+            warnings.warn("Input internal strain tensor does "
+                          "not satisfy standard symmetries")
+        return obj.view(cls)
+
+class ForceConstantMatrix(Tensor):
+    """
+    This class describes the NxNx3x3 born effective charge tensor
+    """
+
+    def __new__(cls, input_array, tol=1e-3):
+        """
+        Create an ForceConstantMatrix object.  The constructor throws an error if
+        the shape of the input_matrix argument is not NxNx3x3, i. e. in true
+        tensor notation. Note that the constructor uses __new__ rather than
+        __init__ according to the standard method of subclassing numpy
+        ndarrays.
+
+        Args:
+            input_matrix (NxNx3x3 array-like): the NxNx3x3 array-like
+                representing the force constant matrix
+        """
+        obj = super(PiezoTensor, cls).__new__(cls, input_array, check_rank=4)
+
+        return obj.view(cls)
+
+
+def get_rand_BEC(struc, eqop, symops, max_charge=1):
+    """
+    Generate a random born effective charge tensor which obeys a structure's 
+    symmetry and the acoustic sum rule
+
+    Args:
+        struc (pymatgen structure): 
+        eqop: operations which map atoms to equivalent atoms 
+            in the structure
+        symops: point symmetry operations of each atomic site
+        max_charge (float): maximum born effective charge value
+
+    Return:
+        BornEffectiveChargeTensor object
+    """
+
+    symstruc = sga(struc)
+    maptoprim = sgastruc.get_symmetry_dataset()['mapping_to_primitive']
+    eqatoms = sgastruc.get_symmetry_dataset()['equivalent_atoms']
+    
+    l = len(struc)
+    BEC = np.zeros((l,3,3))
+    primsites = []
+    for i in range(l):
+        good = 0    
+        for j in reversed(range(i)):
+            if maptoprim[j] == maptoprim[i]:
+                BEC[i] = BEC[j]
+                good = 1
+                break
+            if eqatoms[j] == eqatoms[i] and maptoprim[j] != maptoprim[i]:
+                for k in range(len(eqop[i])):
+                    if eqop[i][k][0] == j:
+                        tempbec = BEC[j]
+                        for m in range(eqop[i][k][2]):
+                            tempbec = eqop[i][k][1].transform_tensor(tempbec)
+                            good = 1
+
+                        
+                        BEC[i] = tempbec
+                    if good == 1:
+                        break
+            if good == 1:
+                break
+        if good == 1:
+            continue
+                
+                
+        symsite = 0 
+        temp_tensor = Tensor(np.random.rand(3,3)-0.5)         
+        temp_tensor = sum([temp_tensor.transform(symm_op) for symm_op in symops[i]]) \
+            /len(symops[i])
+            
+        BEC[i] = np.array(temp_tensor)
+
+    count = 0
+
+    
+#     Enforce Acoustic Sum
```

Comment:
These should be submethods of the appropriate class as well. Then you can just call it. This makes the code a bit cleaner to read.